<a href="https://colab.research.google.com/github/tracieschroeder/DAP-Activities/blob/main/Mass_of_US_Pennies_DAP_Student_Level_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This is a Level 1 notebook. It is designed for students who are not proficient in coding, but gives them to opportunity to review the code and make small changes such as axis names and color. Students are required to enter data and not much else. Histograms are already coded and automatically generated.

#Mass of US Pennies

Particle physicists use graphs like the one shown to look at the results of their experiments. By putting mass readings on a histogram, they can see that the peaks show separate particles.

![histogram](https://github.com/tracieschroeder/CodingCamp/raw/main/histogram.png)

Since we don’t have ready access to particles or machines to make them, we are going to look at something more readily available: U.S. pennies. There are lots of pennies in circulation.

##Learning Check
- Are they all the same?
- They all represent $0.01 and may be similar in color, but is that the only thing that they have in common?

#Obtain 10 Pennies
Begin by brainstorming characteristics of a U.S. penny. Identify the characteristics that can be measured or described with a number value. Measure and record as many properties of each penny as you can observe. Be sure the mass in grams is one of them!

Find the mass of each penny to the nearest 0.1 gram. Write this mass onto a sticky note.

Sort your sticky notes into "bins" so that it makes a “frequency” graph with masses in 0.1 g bins and the number of pennies that fall in each bin. This is a histogram showing the number of pennies in each mass range.

##Learning Check

- Do you see any patterns?
- What do you think you could conclude about the massess of the pennies?

#We can also use this data to create a digital histogram.

Use your group data to enter the mass of each penny into a separate box below.

In [ ]:
A1 = None # @param {type:"number"}
A2 = None # @param {type:"number"}
A3 = None # @param {type:"number"}
A4 = None # @param {type:"number"}
A5 = None # @param {type:"number"}
A6 = None # @param {type:"number"}
A7 = None # @param {type:"number"}
A8 = None # @param {type:"number"}
A9 = None # @param {type:"number"}
A10 = None # @param {type:"number"}

In [ ]:
import matplotlib.pyplot as plt
import numpy as np

A=[A1,A2,A3,A4,A5,A6,A7,A8,A9,A10]

bin_list = np.array([2.000, 2.500, 3.000, 3.500, 4.000])

plt.hist(A,bins=np.subtract(bin_list,0.05), color='darkseagreen', edgecolor = 'black', log=False)
plt.title("What are we measuring?")
plt.xlabel("mass (g)")
plt.ylabel("number of events")
plt.grid(False);

NameError: name 'A1' is not defined

It looks like there is a peak in the middle of our histogram.

##Learning Check

- What is the mass range of the bar in the middle?

- How many pennies have that mass range?

- How does your digital histogram compare to your sticky note histogram?

We measured our pennies to the nearest 0.1 g. It might be a good idea to show our data in terms of the number of pennies for that mass range as well.

Let's change the size of our bin width from 0.5 to 0.1.

In [ ]:
A=[A1,A2,A3,A4,A5,A6,A7,A8,A9,A10]

myBins = []
i = 0
while (i <=4):
  myBins.append(round(i,1))
  i = i + 0.1

bin_list = myBins

plt.hist(A,bins=np.subtract(bin_list,0.01), color='red', edgecolor = 'black', log=False)
plt.title("What are we measuring?")
plt.xlabel("mass (g)")
plt.ylabel("number of events")
plt.grid(False);

##Learning Check

- What is the mass range of the bar in the middle?

- How many pennies have that mass range?

- How does your digital histogram compare to your sticky note histogram?

#More Data
This histogram was made with only ten data points. While we can draw some conclustions based on these results, we can often get a better picture if we add more data.

Now make a histogram using all of the class data.

In the first line, after the A10, but inside the bracket, type in all of the data that the other groups in the class collected. Separate each number with a comma.

In [ ]:
All=[A1,A2,A3,A4,A5,A6,A7,A8,A9,A10, type the other results here!]

myBins = []
i = 0
while (i <=4):
  myBins.append(round(i,1))
  i = i + 0.1

bin_list = myBins
plt.xlim(2,4)
plt.hist(All,bins=np.subtract(bin_list,0.01), color='goldenrod', edgecolor = 'black', log=False)
plt.title("What are we measuring?")
plt.xlabel("mass (g)")
plt.ylabel("number of events")
plt.grid(False);

##Learning Check

Answer the following questions on another sheet of paper. Show all calculations.
1. Describe the masses of the pennies in your set. Be as specific as you can.
2. Describe the masses of the pennies in the class set. Are there any differences between your set of data and the class set?
3. Can you suggest an explanation for the mass distribution that you see? What evidence can you provide to support your explanation?
4. Are there more pennies from more recent years or from prior years? Can you suggest an explanation for this? How would you test this idea?
5. Which year is represented by the most pennies? By the second most? By the third most?
6. What conclusion can you draw based on the evidence provided by your two histograms?
7. What additional measurements might you make to learn more? What tools might you need?

#Lots of Data

You may have noticed that the more sticky notes you added to your histogram, the more patterns began to emerge. Let's see if we can get an even better picture with a very large data set.

First, we will need to import a few libraries. We need pandas so we can read our data file.

In [ ]:
import pandas as pd
import matplotlib as mpl

The next thing we need is our data. The information we want to use is stored as a csv file in excel. We want to import this into our notebook.

The pandas library is what allows us to be able to read the csv file stored on github.

In [ ]:
data = pd.read_csv("https://github.com/tracieschroeder/data-files/raw/main/Penny_Mass_Data_Set.csv")

##Learning Check
- What did we name our data set?

Where is our data? Why can't we see anything?

We haven't told the computer to print it yet. Remove the hashtag below and play that cell.

In [ ]:
#data.head(5)

In [ ]:
data.shape

(1061, 3)

Look at the two numbers in parentheses just below the data.shape command. There are 1061 rows of data in this set. Why did the computer not print them all?

Go back up and change the command to data.head(10) after the data command.

What happens to our data table?

#One More Histogram

Matplotlib is the library we are going to use to create a histogram of our penny data.

Notice the difference in the command used to make this histogram. What do you think might be missing in this iteration?


In [ ]:
myBins = []
i = 0
while (i <=4):
  myBins.append(round(i,1))
  i = i + 0.1

bin_list = myBins

plt.hist(data['Mass'],bins=np.subtract(bin_list,0.01), color='purple', edgecolor = 'black')

plt.xlim(2,4)
plt.title("What are we measuring?")
plt.xlabel("mass (g)")
plt.ylabel("number of events")
plt.grid(False);

If we go back up to our data table, we see we have three columns. Year, Mass, Mint. We could plot any of these, but for this round, we went with Mass. Instead of plotting mass, we could also plot the year or the location. Do either of these things correlate to the mass? What might that say about the mass of the penny?

Try changing the number of bins to 20. How does that change each peak location? How does it change the number of events for each bin?

##Extension
If other characteristics were recorded, you might construct histograms for these results. In some cases, you
may get more insight by trying “scatter plots” or A vs B type graphs. A graph of mass as a function of one
of the other quantities you measured would be a scatter plot. What conclusion can you draw from your new
plots? Did any of the new plots contribute to your conclusions? How?

This notebook was designed by [Quarknet](https://quarknet.org/) Coding Fellow [Tracie Schroeder](https://github.com/tracieschroeder) with a couple crucial coding hints from Mike Plucinkski.




Shield: [![CC BY-NC-SA 4.0][cc-by-nc-sa-shield]][cc-by-nc-sa]

This work is licensed under a
[Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License][cc-by-nc-sa].

[![CC BY-NC-SA 4.0][cc-by-nc-sa-image]][cc-by-nc-sa]

[cc-by-nc-sa]: http://creativecommons.org/licenses/by-nc-sa/4.0/
[cc-by-nc-sa-image]: https://licensebuttons.net/l/by-nc-sa/4.0/88x31.png
[cc-by-nc-sa-shield]: https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-lightgrey.svg
